code/modules/hippie/pool/pool_controller.dm

```diff
@@ -0,0 +1,363 @@
+
+//Originally stolen from paradise. Credits to tigercat2000.
+//Modified a lot by Kokojo.
+/obj/machinery/poolcontroller
+	name = "Pool Controller"
+	desc = "A controller for the nearby pool."
+	icon = 'icons/turf/pool.dmi'
+	icon_state = "poolcnorm"
+	anchored = 1
+	density = 1
+	use_power = 1
+	idle_power_usage = 75
+	var/list/linkedturfs = list() //List contains all of the linked pool turfs to this controller, assignment happens on New()
+	var/temperature = "normal" //The temperature of the pool, starts off on normal, which has no effects.
+	var/srange = 6 //The range of the search for pool turfs, change this for bigger or smaller pools.
+	var/linkedmist = list() //Used to keep track of created mist
+	var/misted = 0 //Used to check for mist.
+	var/obj/item/weapon/reagent_containers/beaker = null
+	var/cur_reagent = "water"
```

Comment:
Please turn this into a linked reagent datum instead of a text-named reagent, this way this will actually fail to compile if we ever remove water